core/src/main/kotlin/core/ems/service/exercise/dir/ReadDir.kt

```diff
@@ -224,18 +223,39 @@ class ReadDirController {
                         dir.id.toString(),
                         ex.title,
                         dir.access,
+                        // If current dir is shared, then children is also shared. Root dir is not shared.
+                        if (currentDir?.isShared == true) true else isDirectoryShared(dir.id, caller),
                         ex.graderType,
                         if (ex.usedOnCourse) courseCount else 0,
                         ex.createdAt,
-                        ex.modifiedAt
+                        ex.createdBy,
+                        ex.modifiedAt,
+                        ex.modifiedBy
                     )
                 }
 
             Resp(currentDir, childDirs, childExercises)
         }
     }
 
-    private fun selectThisDir(dirId: Long?, currentDirAccess: DirAccessLevel?): DirResp? {
+    private tailrec fun isDirectoryShared(dirId: Long, caller: EasyUser): Boolean {
+        val callerGroupId = getImplicitGroupFromAccount(caller.id)
+        val accessCountNotZero = GroupDirAccess
```

Comment:
The following 3 queries can be done with 1 query. (Dir leftJoin GroupDirAccess)

Reply:
Also, this still doesn't work for admin... all private dirs (created by teachers) would show as shared right now. I think counting accesses might work - if there's 
- 0 accesses - teachers don't have access to this dir, so not shared
- 1 access - one teacher has access (in addition to admins), so not shared
- >1 access - is shared
```kotlin
val hasMultipleAccess = GroupDirAccess
            .select { GroupDirAccess.dir eq dirId }
            .count() > 1
```
But should groupBy count to use 1 query instead of 3.